crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1189 @@
+                            })),
+                            cons: Box::new(Stmt::Block(BlockStmt {
+                                span: DUMMY_SP,
+                                stmts: vec![AssignExpr {
+                                    span: DUMMY_SP,
+                                    op: op!("="),
+                                    left: PatOrExpr::Expr(Box::new(
+                                        export_obj.clone().computed_member(key_ident.clone()),
+                                    )),
+                                    right: Box::new(target.computed_member(key_ident)),
+                                }
+                                .into_stmt()],
+                            })),
+                            alt: None,
+                        })],
+                    })),
+                }));
+
+                while !meta.export_names.is_empty() {
+                    meta.setter_fn_stmts.push(
+                        AssignExpr {
+                            span: DUMMY_SP,
+                            op: op!("="),
+                            left: PatOrExpr::Expr(Box::new(Expr::Member(MemberExpr {
+                                span: DUMMY_SP,
+                                obj: Box::new(Expr::Ident(export_obj.clone())),
+                                prop: MemberProp::Ident(quote_ident!(meta.export_names.remove(0))),
+                            }))),
+                            right: meta.export_values.remove(0),
+                        }
+                        .into_stmt(),
+                    );
+                }
+                meta.setter_fn_stmts.push(
+                    CallExpr {
+                        span: DUMMY_SP,
+                        callee: Callee::Expr(Box::new(Expr::Ident(quote_ident!("_export")))),
+                        args: vec![Expr::Ident(export_obj).as_arg()],
+                        type_args: None,
+                    }
+                    .into_stmt(),
+                );
+            }
+        }
+        meta.setter_fn_stmts
+    }
+
+    fn add_module_item_meta(&mut self, mut m: ModuleItemMeta) {
+        match self
+            .module_item_meta_list
+            .iter()
+            .position(|i| m.src.eq(&i.src))
+        {
+            Some(index) => {
+                let mut meta = self.module_item_meta_list.remove(index);
+                meta.setter_fn_stmts.append(&mut m.setter_fn_stmts);
+                meta.export_names.append(&mut m.export_names);
+                meta.export_values.append(&mut m.export_values);
+                if m.has_export_all {
+                    meta.has_export_all = m.has_export_all;
+                }
+                self.module_item_meta_list.insert(index, meta);
+            }
+            None => {
+                self.module_item_meta_list.push(m);
+            }
+        }
+    }
+
+    fn hoist_var_decl(&mut self, var_decl: VarDecl) -> Option<Expr> {
+        let mut exprs = vec![];
```

Comment:
I think type of this variable should be `Vec<Box<Expr>>`, to remove needlesss `memmove`s.

Note that you don't need to annotate type.